saas/mic-hex-escs/domain/src/main/java/com/sokima/saas/mic/hex/escs/domain/transformer/clean/mapper/Tuple2CleanedCartMapper.java

```diff
@@ -0,0 +1,36 @@
+package com.sokima.saas.mic.hex.escs.domain.transformer.clean.mapper;
+
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.Multiset;
+import com.google.common.collect.Multisets;
+import com.sokima.lib.ecommerce.java.domain.model.Cart;
+import com.sokima.lib.ecommerce.java.domain.model.record.CartRecord;
+import com.sokima.saas.mic.hex.escs.domain.payload.clean.CleanCartPayload;
+
+import java.util.List;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+import org.mapstruct.factory.Mappers;
+
+import reactor.util.function.Tuple2;
+
+@Mapper
+public interface Tuple2CleanedCartMapper {
+
+    Tuple2CleanedCartMapper INSTANCE = Mappers.getMapper(Tuple2CleanedCartMapper.class);
+
+    @Mapping(target = "cartId", source = "t1.cartId")
+    @Mapping(target = "userId", expression = "java(input.getT2().userId())")
```

Comment:
use java expression only when any supported features are not working. 
here you can use source = "t2.userId" like you did it for previous mapping. 
```suggestion
    @Mapping(target = "userId", source= "t2.userId")
```